### PR TITLE
V2 podman system connection

### PR DESCRIPTION
**Pull request by jwhonce**
* Implement command
* Refactor podman to pull from containers.conf/connections.conf by default
* Write podman-system-connection.1.md

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: cmd/podman/registry/config.go [hunk 1]
```diff
@@ -68,22 +68,23 @@ func newPodmanConfig() {
...
 		fmt.Fprint(os.Stderr, "Failed to obtain podman configuration: "+err.Error())
 		os.Exit(1)
 	}
 
-	cfg.Network.NetworkConfigDir = cfg.Network.CNIPluginDirs[0]
-	if rootless.IsRootless() {
-		cfg.Network.NetworkConfigDir = ""
+	cfg.Network.NetworkConfigDir = ""
+	if !rootless.IsRootless() {
+		if len(cfg.Network.CNIPluginDirs) > 0 {
```
Comment (mheon) [inline cmd/podman/registry/config.go:79]: Why this change? CNI should never enter the picture in rootless mode

File: cmd/podman/registry/config_tunnel.go [hunk 2]
```diff
@@ -2,6 +2,13 @@
...
 
+import (
+	"os"
+)
+
 func init() {
 	abiSupport = false
+
+	// Enforce that podman-remote == podman --remote
+	os.Args = append(os.Args, "--remote")
```
Comment (mheon) [inline cmd/podman/registry/config_tunnel.go:13]: Is there a potential problem if we duplicate this? Should we only append if --remote is not already set?

File: cmd/podman/root.go [hunk 3]
```diff
@@ -103,13 +103,22 @@ func persistentPreRunE(cmd *cobra.Command, args []string) error {
...
-
 	// Help is a special case, no need for more setup
 	if cmd.Name() == "help" {
 		return nil
 	}
 
+	cfg := registry.PodmanConfig()
+	if cfg.Remote {
+		if !cmd.Flag("url").Changed {
+			_ = cmd.Flags().Set("url", cfg.Config.Engine.RemoteURI)
```
Comment (rhatdan) [inline cmd/podman/root.go:114]: Why not just default to these values, and only use them if user specifies remote.  That way the user will see them if he does podman run --help?

File: cmd/podman/registry/config.go (hunk 1 above)
Comment (jwhonce) [reply cmd/podman/registry/config.go:79]: During testing with the remote client, the previous code panic'ed on the [0]

File: cmd/podman/registry/config_tunnel.go (hunk 2 above)
Comment (jwhonce) [reply cmd/podman/registry/config_tunnel.go:13]: duping, doesn't hurt. This is faster than searching the list and appending.

File: cmd/podman/root.go (hunk 3 above)
Comment (jwhonce) [reply cmd/podman/root.go]: D*mn, and I even had a TODO to implement this way.  Fixed.

```
$ bin/podman --help |grep identity                                                    
      --identity string           path to SSH identity file, (CONTAINER_SSHKEY) (default "/home/jhonce/.ssh/ocp-tier3") 
```

File: cmd/podman/registry/config.go [hunk 4]
```diff
@@ -68,22 +68,23 @@ func newPodmanConfig() {
 		}
 	}
 
-	// FIXME: for rootless, add flag to get the path to override configuration
-	cfg, err := config.NewConfig("")
+	cfg, err := config.ReadCustomConfig()
```
Comment (rhatdan) [inline cmd/podman/registry/config.go:71]: This should not change.  This eliminates all of the defaults and reads only a custom config.  It should never be done for the global setting, only when you are customising the local.
This is why SELinux is breaking.

**Comment by rhatdan** [comment]
LGTM
